js/app.js

```diff
@@ -1,5 +1,41 @@
-angular.module('GiphyApp', [])
+angular.module('GiphyApp', ['infinite-scroll'])
 
-.run(function() {
-  console.log('App has loaded!');
-});
+.controller('myCtrl', ['$scope', '$http', function($scope, $http){
+  $scope.offset = 0;
+  $scope.busy = false;
+  $scope.search = 'dogs';
+
+  $scope.add = function(){
+    $scope.results = null;
+    query();
+  }
+
+  function query(){
+    $scope.busy = true;
+
+    $http.get('http://api.giphy.com/v1/stickers/search', {
+      params: {
+        api_key: 'dc6zaTOxFJmzC',
+        q: $scope.search,
+        limit: 30,
+        offset: $scope.offset
+      }
+    })
+    .then(function success(res){
+      if(!$scope.results) {
+        $scope.results = res.data.data;
+      } else {
+        Array.prototype.push.apply($scope.results, res.data.data);
+      }
+      $scope.busy = false;
+    }, function error(res){
+      console.log(res);
+    });
+  }
+
+  $scope.next = function(){
+    console.log("looking for more stuff");
+    $scope.offset += 30;
+    query();
+  }
+}]);
```

Comment:
Just for funsies here is some code from my deliverable to handle the infinite scrolling:

    $scope.search = function () {
        if ($scope.searchTerm != $scope.oldSearchTerm) {$scope.giphys = [];};
        var req = {
            url: "http://api.giphy.com/v1/gifs/search",
            method: 'GET',
            params: {
                api_key: $scope.api,
                offset: $scope.offset,
                q: $scope.searchTerm,
            }
        }
        $http(req).then(function success(res) {
            $scope.giphys = $scope.giphys.concat(res.data.data);
        }, function error (res) {
            console.log(res);
        });
    };

    $scope.myPagingFunction = function (){
        $scope.oldSearchTerm = $scope.searchTerm;
        $scope.offset = 25 * $scope.count;
        $scope.search();
        $scope.count++;
    };